### DISCUSSION TITLE
Auto hide & show deeper Contents sidebar levels.

### DISCUSSION THREAD

**Discussion by KasperLin** [pradyunsg/furo#600] [Ideas]
As #566 has shown, [furo](https://pradyunsg.me/furo/) user can hide deeper *"Contents Sidebar (right)"* levels. 

I wonder if it's possible for furo to have those deeper hidden levels automatically reveal themselves, corresponding to user's scrolling / clicking, just like [this page](https://pandas.pydata.org/docs/user_guide/io.html) does via [pydata-sphinx-theme](https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/page-toc.html#show-more-levels-of-the-in-page-toc-by-default). 

> Several other sphinx theme supports this, such as [Jupyter Book](https://jupyterbook.org/en/stable/intro.html). 

Can we have this feature ? Is it technically possible ? I believe this can be a beautiful feature for furo and match [it's design philosophy](https://pradyunsg.me/furo/contributing/design/#guiding-ideas) well.

**Comment by pradyunsg** [comment]
> Is it technically possible?

Yes.

> Can we have this feature?

Hmm... I'm not keen on adding knobs to Furo for stuff like this.

In that case, this will end up being a case of "should we hide the structure of in-page content from readers". I'm wary of that as well.

It seems unlikely to me; but I could be convinced that Furo should have a knob for this.
